### PR TITLE
add user/update

### PR DESCRIPTION
**Pull request by haubitzer**

### CODE REVIEW

File: src/users/users.test.ts [hunk 1]
```diff
@@ -53,4 +56,10 @@ describe('Users', () => {
     resultSingle = users.readById(1);
     await resultSingle.then((result) => expect(result).toStrictEqual({}));
   });
+
+  test('update', async () => {
+    mock.onPut(updatePath).reply(200, { Success: true, Results: [{}] });
+    resultUpdateRaw = users.update({ id: 1 });
+    await resultUpdateRaw.then((result) => expect(result).toMatchObject({ data: {} }));
```
Comment (benj56) [inline src/users/users.test.ts:63]: I see that it's the same in the test above but I wouldn't do this, there is no need for `then` when there is `await`. So `expect` can be outside of the arrow function:
```suggestion
    expect(await resultUpdateRaw).toMatchObject({ data: {} });
```

File: src/utils/response/resourceResponse.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ export type ResourceResponse<T> = {
   success: boolean;
   results: T[];
   deleted: DeletedEntry[];
-  affected?: Record<string, any[]>;
+  affected: Record<string, any[]>; // TODO this should not be an record, it is should be AffectedBy<T>
```
Comment (benj56) [inline src/utils/response/resourceResponse.ts:12]: Just on a side note: `any` isn't allowed with updated ESLint rules, there will be a bigger refactoring of response types to get that in proper shape: https://github.com/TimeTac/js-client-library/pull/82/files#diff-9b27af540065038ee20390203cc238464beb9804ee7974ebf3ee307894e817ea
Also `DeltaSyncResponse` is interesting I think. Affected is probably still wrong in that PR, I think it's missing an Array. We can talk about it also in the meeting with @whosshell

**Review by benj56 — DISMISSED** [review]

File: src/utils/response/resourceResponse.ts (hunk 2 above)
Comment (haubitzer) [reply src/utils/response/resourceResponse.ts:12]: FYI: affected is returned as an object by the backed, similar to DeltaSyncResponse.

**Review by benj56 — APPROVED** [review]

File: src/utils/response/resourceResponse.ts (hunk 2 above)
Comment (benj56) [reply src/utils/response/resourceResponse.ts:12]: Yes, probably the type would be 
```
  type Affected<R extends Resource> = {
    [r in R]: Entity<r>[]
  }

  type Resource = keyof Resources & string;
  type Entity<R extends Resource> = Resources[R];
  type Resources = {
    timeTrackings: TimeTracking;
    users: User,
  }
```

**Pull request merged**